src/api/rpc_api/author.rs

```diff
@@ -174,11 +188,52 @@ pub trait SubmitAndWatchUntilSuccess {
 	/// - last known extrinsic (transaction) status
 	/// - associated events of the extrinsic
 	/// This method is blocking.
+	#[deprecated(
+		since = "0.14.0",
+		note = "please use `submit_and_watch_opaque_extrinsic_until` instead, this will be removed in the next release."
+	)]
 	async fn submit_and_watch_opaque_extrinsic_until_success(
 		&self,
 		encoded_extrinsic: &Bytes,
 		wait_for_finalized: bool,
 	) -> Result<ExtrinsicReport<Self::Hash>>;
+
+	/// Submit an extrinsic and watch it until the desired status
+	/// is reached, if no error is encountered previously.
+	/// When status is InBlock or Finalized, the triggered events are listed in the report.
+	/// Returns and error if the extrinsic was not successfully executed.
```

Comment:
Suggestion: Split description in two parts:

Something like this:
```rust
/// Submit an extrinsic and watch it until the desired status
/// is reached, if no error is encountered previously.
/// If watched until `InBlock` or `Finalized` this function will either return an Error
/// if the extrinsic execution has failed onchain. 
/// If it was successful, a report containing the following is returned:
/// - extrinsic hash
/// - hash of the block the extrinsic was included in
/// - last known extrinsic (transaction) status
/// - associated events of the extrinsic
/// If not watched until atleast `InBlock` this function will not know if the extrinsic
/// was executed onchain or not and will therefore not return an error upon failed execution.
/// An Error will be returned if the extrinsic has failed to be sent or was not 
/// included into the transaction pool of the node.
/// If no error is enocuntered, a report containing the following is returned:
/// - extrinsic hash
/// - last known extrinsic (transaction) status

```

Reply:
Nice to add the errors differences